src/proto/console/text/output.rs

```diff
@@ -125,20 +125,20 @@ impl<'boot> Output<'boot> {
     /// Sets the cursor's position, relative to the top-left corner, which is (0, 0).
     ///
     /// This function will fail if the cursor's new position would exceed the screen's bounds.
-    pub fn set_cursor_position(&mut self, column: usize, row: usize) -> Result<()> {
+    pub fn set_cursor_position(&mut self, column: usize, row: usize) -> Result {
         (self.set_cursor_position)(self, column, row).into()
     }
 
     /// Sets the text and background colors for the console.
     ///
     /// Note that for the foreground color you can choose any color.
     /// The background must be one of the first 8 colors.
-    pub fn set_color(&mut self, foreground: Color, background: Color) -> Result<()> {
+    pub fn set_color(&mut self, foreground: Color, background: Color) -> Result {
         let fgc = foreground as usize;
         let bgc = background as usize;
 
         if bgc >= 8 {
-            Err(Status::DEVICE_ERROR)
+            Status::INVALID_PARAMETER.into()
```

Comment:
I think it makes more sense to classify bad background color values as invalid parameters rather than as device errors.

Reply:
Considering that correctly written applications wouldn't pass an invalid background color to this function, should we replace this check with an assertion?

Reply:
Makes sense ! In an ideal world, we could [use the type system to clarify that background colors are a subset of foreground ones](https://en.wikipedia.org/wiki/Refinement_type), but that is currently too much boilerplate in Rust :)